v6_MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,226 @@
+
+## Table of Contents
+
+1. [Gradle](#gradle)
+1. [Builder Pattern](#builder-pattern)
+1. [DropInRequest](#dropinrequest)
+1. [Authorization](#authorization)
+1. [Launch Drop-In](#launch-drop-in)
+1. [Handle Drop-In Result](#handle-drop-in-result)
+1. [Fetch Last Used Payment Method](#fetch-last-used-payment-method)
+
+## Gradle
+
+Add the dependency in your `build.gradle`:
+
+```groovy
+dependencies {
+  implementation 'com.braintreepayments.api:drop-in:6.0.0'
+}
+```
+
+The credentials for accessing the Cardinal Mobile SDK have changed.
+
+Update the credentials in your top-level build.gradle to the following:
+
+```groovy
+repositories {
+    maven {
+        url "https://cardinalcommerceprod.jfrog.io/artifactory/android"
+        credentials {
+            username 'braintree_team_sdk'
+            password 'AKCp8jQcoDy2hxSWhDAUQKXLDPDx6NYRkqrgFLRc3qDrayg6rrCbJpsKKyMwaykVL8FWusJpp'
+        }
+    }
+}
+```
+
+## Builder Pattern
+
+The builder pattern has been removed in v6 to allow for consistent object creation across Java and Kotlin. 
+Method chaining has been removed, and setters have been renamed with the `set` prefix.
+
+For example, a `DropInRequest` can now be constructed below:
+
+Java:
+```java
+DropInRequest request = new DropInRequest();
+request.setCollectDeviceData(true);
+request.setMaskCardNumber(true);
+```
+
+Kotlin:
+```kotlin
+val request = DropInRequest()
+request.collectDeviceData = true
+request.maskCardNumber = true
+```
+
+See the [Braintree Android v4 Migration Guide](https://github.com/braintree/braintree_android/blob/master/v4_MIGRATION_GUIDE.md) for changes to request objects for Google Pay, Venmo, PayPal, and 3DS. 
+
+## DropInRequest
+
+The getters and setters on `DropInRequest` have been renamed with a consistent get/set pattern to allow for Kotlin synthesized properties. 
+See the examples below for a full list of optional parameters:
+
+Java:
+```java
+    DropInRequest dropInRequest = new DropInRequest();
+    dropInRequest.setShouldCollectDeviceData(true);
+    dropInRequest.setGooglePayRequest(googlePayRequest);
+    dropInRequest.setGooglePayDisabled(true);
+    dropInRequest.setPayPalRequest(paypalRequest);
+    dropInRequest.setPayPalDisabled(true);
+    dropInRequest.setVenmoDisabled(true);
+    dropInRequest.setCardDisabled(true);
+    dropInRequest.setShouldRequestThreeDSecureVerification(true);
+    dropInRequest.setThreeDSecureRequest(threeDSecureRequest);
+    dropInRequest.setShouldMaskCardNumber(true);
+    dropInRequest.setShouldMaskSecurityCode(true);
+    dropInRequest.setEnableVaultManager(true);
+    dropInRequest.setAllowVaultCardOverride(true);
+    dropInRequest.setVaultCardDefaultValue(true);
+    dropInRequest.setCardholderNameStatus(CardForm.FIELD_OPTIONAL);
+    dropInRequest.setVaultVenmoDefaultValue(true);
+```
+
+Kotlin:
+```kotlin
+    val dropInRequest = DropInRequest()
+    dropInRequest.shouldCollectDeviceData = true
```

Comment:
In iOS Drop In - we are always returning device data now on the DropInResult. So we don't have this param on the DropInRequest. We can also make this change for Android Drop In v6 while we have the luxury of making breaking changes 👯

Reply:
Nice ok I'll create a JIRA card for that change before GA 👍